### PR TITLE
Create certificate if certificateArn is omitted.

### PR DESCRIPTION
**Pull request by clstokes**
If certificateArn is omitted from the configuration, create an ACM certificate and associated DNS records for certificate verification.

### CODE REVIEW

File: aws-ts-static-website/index.ts [hunk 1]
```diff
@@ -79,6 +78,38 @@ const logsBucket = new aws.s3.Bucket("requestLogs",
...
+    });
+    const certificate = new aws.acm.Certificate("certificate", {
+        domainName: config.targetDomain,
+        validationMethod: "DNS",
+    }, { provider: eastRegion });
+
+    const domainParts = getDomainAndSubdomain(config.targetDomain);
+    const hostedZoneId = aws.route53.getZone({ name: domainParts.parentDomain }).then(zone => zone.id);
+
+    const certificateValidationDomain = new aws.route53.Record(
```
Comment (chrsmith) [inline aws-ts-static-website/index.ts]: A comment to better explain what is going on here is probably warranted. What you are doing is creating the ACM certificate, that resource then has the `domainValidationOptions`, which you then use to create a new DNS record, which is used to verify that you own the domain. Right?

It's cool that it all "just works", but some docs explaining what's going on would be useful for the next person who looks at this code scratching their head.

File: aws-ts-static-website/index.ts [hunk 2]
```diff
@@ -79,6 +78,38 @@ const logsBucket = new aws.s3.Bucket("requestLogs",
...
+        certificateArn: certificate.arn,
+        validationRecordFqdns: [certificateValidationDomain.fqdn],
+    }, { provider: eastRegion });
+
+    certificateArn = certificateValidation.certificateArn;
+}
+
 // distributionArgs configures the CloudFront distribution. Relevant documentation:
 // https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/distribution-web-values-specify.html
 // https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html
```
Comment (chrsmith) [inline aws-ts-static-website/index.ts]: Would you mind adding a comment on top of the `aliases` attribute here too, just to make the next time this example is built upon:

```diff
 const distributionArgs: aws.cloudfront.DistributionArgs = {
     enabled: true,	     enabled: true,
+    // Alternate aliases the CloudFront distribution can be reached at, in addition to https://xxxx.cloudfront.net.
+    // Required if you want to access the distribution via config.targetDomain as well.  
     aliases: [ config.targetDomain ],
```

File: aws-ts-static-website/index.ts (hunk 1 above)
Comment (clstokes) [reply aws-ts-static-website/index.ts]: Yep, great point.

File: aws-ts-static-website/index.ts (hunk 2 above)
Comment (clstokes) [reply aws-ts-static-website/index.ts]: Added!

**Comment by clstokes** [comment]
@chrsmith, added some additional comments. If you have a chance, take another quick pass and see if I missed anything, Thanks!

**Pull request merged**